controller/experiment.controller.js

```diff
@@ -0,0 +1,88 @@
+const {json} = require("express");
+const ExperimentRepository = require("../repositories/experiment.repository");
+const {PropertyNotFound} = require("../errors/NotFound.errors");
+const {ServerUnableError} = require("../errors/internal.errors");
+const experimentRepository = new ExperimentRepository();
+
+const getAllExperiments = async (req, res) => {
+    const result = await experimentRepository.find();
+    if (!result) throw new ServerUnableError("getAllExperiments")
+    res.status(200).json({result});
+}
+
+const getExperimentById = async (req, res) => {
+    if (!req.params.experiment_id) throw new PropertyNotFound("experiment_id");
+    const experiment_id = req.params.experiment_id;
+    const result = await experimentRepository.retrieve(experiment_id);
+    if (!result) throw new ServerUnableError("getExperimentById")
+    res.status(200).json({result});
+}
+
+const getExperimentsByAccountId = async (req, res) => {
+    if (req.params.account_id) throw new PropertyNotFound("account_id");
+    const account_id = req.params.account_id;
+    const result = await experimentRepository.findByAccount("account_id", account_id);
+    if (!result) throw new ServerUnableError("getExperimentsByAccountId")
+    res.status(200).json({result});
+}
+
+const getExperimentsAB = async (req, res) => {
+    if (req.params.account_id) throw new PropertyNotFound("account_id");
+    const account_id = req.params.account_id;
+    const result = await experimentRepository.findGroup("type", "a-b", "account_id", account_id);
+    if (!result) throw new ServerUnableError("getExperimentsAB")
+    res.status(200).json({result});
+
+}
+
+const getExperimentsFF = async (req, res) => {
+    if (req.params.account_id) throw new PropertyNotFound("account_id");
+    const account_id = req.params.account_id;
+    const result = await experimentRepository.findGroup("type", "f-f", "account_id", account_id);
+    if (!result) throw new ServerUnableError("getExperimentsFF")
+    res.status(200).json({result});
+
+}
+
+const getExperimentsByDate = async (req, res) => {
+    if (req.query.year && req.query.month) throw new PropertyNotFound("year and month");
```

Comment:
better do this separately for year and month individually.
also, forgot the "!" on both